test-utils/src/test_indexer.rs

```diff
@@ -0,0 +1,428 @@
+                in_path_indices: BigInt::from_be_bytes(leaf_index.to_be_bytes().as_slice()), // leaf_index as u32,
+                in_path_elements: proof.iter().map(|x| BigInt::from_be_bytes(x)).collect(),
+            });
+        }
+
+        let inclusion_proof_inputs = InclusionProofInputs(inclusion_proofs.as_slice());
+        let json_payload =
+            InclusionJsonStruct::from_inclusion_proof_inputs(&inclusion_proof_inputs).to_string();
+
+        let response_result = client
+            .post(&format!("{}{}", SERVER_ADDRESS, INCLUSION_PATH))
+            .header("Content-Type", "text/plain; charset=utf-8")
+            .body(json_payload)
+            .send()
+            .await
+            .expect("Failed to execute request.");
+        assert!(response_result.status().is_success());
+        let body = response_result.text().await.unwrap();
+        let proof_json = deserialize_gnark_proof_json(&body).unwrap();
+        let (proof_a, proof_b, proof_c) = proof_from_json_struct(proof_json);
+        let (proof_a, proof_b, proof_c) = compress_proof(&proof_a, &proof_b, &proof_c);
+
+        let merkle_tree_account =
+            AccountZeroCopy::<StateMerkleTreeAccount>::new(context, self.merkle_tree_pubkey).await;
+        let merkle_tree = merkle_tree_account
+            .deserialized()
+            .copy_merkle_tree()
+            .unwrap();
+        assert_eq!(
+            self.merkle_tree.root(),
+            merkle_tree.root().unwrap(),
+            "Local Merkle tree root is not equal to latest onchain root"
+        );
+
+        let root_indices: Vec<u16> =
+            vec![merkle_tree.current_root_index as u16; compressed_accounts.len()];
+        (
+            root_indices,
+            CompressedProof {
+                a: proof_a,
+                b: proof_b,
+                c: proof_c,
+            },
+        )
+    }
+
+    /// deserializes an event
+    /// adds the output_compressed_accounts to the compressed_accounts
+    /// removes the input_compressed_accounts from the compressed_accounts
+    /// adds the input_compressed_accounts to the nullified_compressed_accounts
+    pub fn add_lamport_compressed_accounts(&mut self, event_bytes: Vec<u8>) {
+        let event_bytes = event_bytes.clone();
+        let event = PublicTransactionEvent::deserialize(&mut event_bytes.as_slice()).unwrap();
+        self.add_event_and_compressed_accounts(event);
+    }
+
+    pub fn add_event_and_compressed_accounts(
+        &mut self,
+        event: PublicTransactionEvent,
+    ) -> Vec<usize> {
+        for compressed_account in event.input_compressed_accounts.iter() {
+            let index = self
+                .compressed_accounts
+                .iter()
+                .position(|x| x.compressed_account == compressed_account.compressed_account)
+                .expect("compressed_account not found");
+            self.compressed_accounts.remove(index);
```

Comment:
```suggestion
            self.compressed_accounts.retain(|x| 
                x.compressed_account != compressed_account.compressed_account
            );
```

Reply:
interesting, makes sense here for simplicity.